src/Scene.cpp

```diff
@@ -82,5 +86,109 @@ void Scene::run_frame()
     for (auto const& component : components_copy)
     {
         component->update();
+        solve_collisions();
+    }
```

Comment:
We solve all collisions for all colliders... for each component? Seems like this should be out of this loop